### PR TITLE
DM-21410: Convert to a metapackage for ci_hsc_gen2 and _gen3.

### PR DESCRIPTION
**Pull request by TallJimbo**
All removed files are (potentially out-of-date) duplicates of files in testdata_ci_hsc or ci_hsc_gen2.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +0,0 @@
-.sconf_temp
```
Comment (timj) [inline .gitignore]: Are you sure that there are no files to be ignored after a build completes?

Comment (TallJimbo) [reply .gitignore]: This now relies on the `eupspkg` default logic for when a repo has nothing in it, which is to do nothing.  And I've (now) checked that `eupspkg -e build` does not produce any files.

Comment (timj) [reply .gitignore]: I *think* `lsst-build` will make a `_build.log` file.

Comment (TallJimbo) [reply .gitignore]: I'll add that just in case.

**Pull request merged**